Public/Src/Engine/Scheduler/Distribution/ExecutionResultSerializer.cs

```diff
@@ -613,5 +621,51 @@ private void WriteFileArtifact(BuildXLWriter writer, FileArtifact file)
             WriteAbsolutePath(writer, file.Path);
             writer.WriteCompact(file.RewriteCount);
         }
+
+        private static void ReadPipProperties(BuildXLReader reader, out IReadOnlyDictionary<string, int> pipProperties)

```

Comment:
>ReadPipProperties [](start = 28, length = 17)

I'm thinking that we should use [http://index/?leftProject=BuildXL.Utilities&leftSymbol=wdncwk257pvr&file=BuildXLWriter.cs&line=338](http://index/?leftProject=BuildXL.Utilities&leftSymbol=wdncwk257pvr&file=BuildXLWriter.cs&line=338)
for reading and writing.
The pipProperties can be turned into a list of key-value pairs first.

Reply:
What is the benefit of that over the methods added in the review, which does not require converting the dictionary to list first?

Reply:
Code reuse.

---
In reply to: [334752748](https://github.com/microsoft/BuildXL/pull/1000#discussion_r334752748) [](ancestors = 334752748)